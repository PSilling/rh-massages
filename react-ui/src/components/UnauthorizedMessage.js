// react imports
import React, { Component } from 'react';

// util imports
import _t from '../utils/Translations.js';

/**
 * Message shown to unauthorized users.
 */
class UnauthorizedMessage extends Component {

  render() {
    return(
      <div>
        <h1>
          { this.props.title }
        </h1>
        <hr />
          <h2>
            { _t.translate('Unauthorized') }
          </h2>
      </div>
    )
  }
}

export default UnauthorizedMessage
