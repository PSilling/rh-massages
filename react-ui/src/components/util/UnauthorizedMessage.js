// react imports
import React from "react";
import PropTypes from "prop-types";

// util imports
import _t from "../../util/Translations";

/**
 * Simple title message shown to unauthorized users.
 */
const UnauthorizedMessage = function UnauthorizedMessage(props) {
  return (
    <div className="mt-3">
      <h2>{props.title}</h2>
      <hr />
      <h3>{_t.translate("Unauthorized")}</h3>
    </div>
  );
};

UnauthorizedMessage.propTypes = {
  /** title shown above unauthorized notification */
  title: PropTypes.string
};

UnauthorizedMessage.defaultProps = {
  title: "Unauthorized"
};

export default UnauthorizedMessage;
