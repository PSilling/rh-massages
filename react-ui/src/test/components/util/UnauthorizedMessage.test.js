// react imports
import React from "react";
import TestRenderer from "react-test-renderer";

// test imports
import UnauthorizedMessage from "../../../components/util/UnauthorizedMessage";
import _t from "../../../util/Translations";

test("renders content with correct props", () => {
  const testRenderer = TestRenderer.create(<UnauthorizedMessage title="test" />);
  const testInstance = testRenderer.root;
  const header = testInstance.findByType("h2");
  const message = testInstance.findByType("h3");
  const treeJSON = testRenderer.toJSON();

  expect(header.props.children).toEqual("test");
  expect(message.props.children).toEqual(_t.translate("Unauthorized"));
  expect(treeJSON).toMatchSnapshot();
});
