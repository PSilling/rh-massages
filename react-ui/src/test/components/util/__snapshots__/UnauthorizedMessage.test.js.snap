// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<div>
  <h2>
    test
  </h2>
  <hr />
  <h3>
    Neautorizov√°no
  </h3>
</div>
`;
